File: docs/source/examples/Threading/Freeze.m
// Get an immutable copy of the realm that can be passed across threads
RLMRealm *frozenRealm = [realm freeze];

RLMResults *dogs = [Dog allObjectsInRealm:realm];

// You can freeze collections
RLMResults *frozenDogs = [dogs freeze];

// You can still read from frozen realms
RLMResults *frozenDogs2 = [Dog allObjectsInRealm:frozenRealm];

Dog *dog = [dogs firstObject];

// You can freeze objects
Dog *frozenDog = [dog freeze];

File: source/examples/Threading/Freeze.swift
// Get an immutable copy of the realm that can be passed across threads
let frozenRealm = realm.freeze()

let dogs = realm.objects(Dog.self)

// You can freeze collections
let frozenDogs = dogs.freeze()

// You can still read from frozen realms
let frozenDogs2 = frozenRealm.objects(Dog.self)

let dog = dogs.first!

// You can freeze objects
let frozenDog = dog.freeze()
